File: backend/static/actions/inventory.js
// ---------------------
// 📦 ESTADO INVENTARIO
// ---------------------
let inventario = {};
let submodo = null;

// ---------------------
// 🧠 FUNCIÓN PRINCIPAL
// ---------------------
export function ejecutarInventario(texto) {

    texto = texto.toLowerCase().trim();

    // ---------------------
    // 🔒 CONFIRMACIÓN SALIDA
    // ---------------------
    if (submodo === "confirmar_salida") {

        if (texto === "si" || texto === "sí") {
            reset();
            return {
                mensaje: "Saliendo de inventario",
                accion: "salir_confirmado"
            };
        }

        if (texto === "no") {
            submodo = null;
            return {
                mensaje: "Continuamos en inventario"
            };
        }

        return { mensaje: "Responde sí o no" };
    }

    // ---------------------
    // 📊 CERRAR INVENTARIO (MOSTRAR RESUMEN)
    // ---------------------
    if (texto.includes("cerrar")) {

        if (Object.keys(inventario).length === 0) {
            return {
                mensaje: "No hay productos en el inventario"
            };
        }

        let resumen = "Resumen del inventario: ";

        for (let producto in inventario) {
            resumen += `${inventario[producto]} ${producto}, `;
        }

        return {
            mensaje: resumen
        };
    }

    // ---------------------
    // 🚪 SALIR
    // ---------------------
    if (texto.includes("salir") || texto.includes("volver")) {

        if (Object.keys(inventario).length > 0) {
            submodo = "confirmar_salida";
            return {
                mensaje: "Tienes productos. ¿Seguro que quieres salir?"
            };
        }

        return {
            mensaje: "Saliendo de inventario",
            accion: "salir_confirmado"
        };
    }

    // ---------------------
    // 📦 AGREGAR PRODUCTO (CON PARSER)
    // ---------------------
    if (
        texto.includes("agrega") ||
        texto.includes("agregar") ||
        texto.match(/^\d+/) // 🔥 CLAVE: permite "20 galletas"
    ) {

        const resultado = parsearProducto(texto);

        if (!resultado) {
            return {
                mensaje: "No entendí el producto. Di por ejemplo: agregar 10 galletas"
            };
        }

        const { cantidad, producto } = resultado;

        inventario[producto] = (inventario[producto] || 0) + cantidad;

        return {
            mensaje: `Agregado ${cantidad} ${producto}`
        };
    }

    // ---------------------
    // ❌ DEFAULT
    // ---------------------
    return {
        mensaje: "Comando no válido en inventario"
    };
}

// ---------------------
// 🧠 PARSER MEJORADO
// ---------------------
function parsearProducto(texto) {

    // elimina palabras innecesarias
    texto = texto
        .replace("agrega", "")
        .replace("agregar", "")
        .replace("de", "")
        .trim();

    const match = texto.match(/(\d+)\s+(.*)/);

    if (!match) return null;

    const cantidad = parseInt(match[1]);
    const producto = match[2].trim();

    if (!producto) return null;

    return { cantidad, producto };
}

// ---------------------
// 🔄 RESET
// ---------------------
function reset() {
    inventario = {};
    submodo = null;
}

File: backend/modules/inventory.py
from services.pdf_generator import generar_pdf

def ejecutar_inventario(estado, comando):
    inventario = estado["inventario"]

    if "submodo" not in estado:
        estado["submodo"] = None

    # ── CONFIRMACIÓN SALIDA ──────────────────────────────────
    if estado["submodo"] == "confirmar_salida":

        if comando.get("tipo") == "confirmar_si":
            estado["inventario"] = {}
            estado["submodo"]    = None
            return {
                "respuesta": "Listo, saliendo.",
                "accion": "salir_confirmado"
            }

        if comando.get("tipo") == "confirmar_no":
            estado["submodo"] = None
            return {"respuesta": "Continuamos."}

        return {"respuesta": "Di sí o no."}

    # ── CERRAR → PDF + RESUMEN ───────────────────────────────
    if comando.get("tipo") == "cerrar":

        if len(inventario) == 0:
            return {"respuesta": "El inventario está vacío."}

        ruta_pdf = generar_pdf(inventario)

        total_productos = len(inventario)
        total_unidades  = sum(inventario.values())

        # Resumen corto para audio
        lista = [f"{c} {p}" for p, c in list(inventario.items())[:5]]
        resumen_audio = ", ".join(lista)
        if total_productos > 5:
            resumen_audio += f" y {total_productos - 5} productos más"

        estado["inventario"] = {}

        return {
            "respuesta": f"{total_productos} productos, {total_unidades} unidades en total. {resumen_audio}. PDF listo.",
            "pdf": True,
            "archivo": ruta_pdf
        }

    # ── SALIR ────────────────────────────────────────────────
    if comando.get("tipo") == "salir":

        if len(inventario) > 0:
            n = len(inventario)
            estado["submodo"] = "confirmar_salida"
            return {
                "respuesta": f"Tienes {n} productos. ¿Salir sin guardar?"
            }

        return {
            "respuesta": "Saliendo.",
            "accion": "salir_confirmado"
        }

    # ── LISTAR ───────────────────────────────────────────────
    if comando.get("tipo") == "listar":

        if len(inventario) == 0:
            return {"respuesta": "Inventario vacío. Empieza agregando productos."}

        lista = [f"{c} {p}" for p, c in inventario.items()]
        return {"respuesta": "Tienes: " + ", ".join(lista)}

    # ── AGREGAR PRODUCTO ─────────────────────────────────────
    if comando.get("tipo") == "agregar":

        producto = comando["producto"]
        cantidad = comando["cantidad"]

        anterior = inventario.get(producto, 0)
        inventario[producto] = anterior + cantidad
        nuevo_total = inventario[producto]

        # Si ya existía el producto, confirmar acumulado
        if anterior > 0:
            return {
                "respuesta": f"{cantidad} {producto}. Total {nuevo_total}."
            }

        return {
            "respuesta": f"{cantidad} {producto}, listo."
        }

    # ── FALLBACK AMIGABLE ────────────────────────────────────
    texto_original = comando.get("texto_original", "")
    if texto_original:
        return {
            "respuesta": f"No entendí. Di el número y el producto, por ejemplo: veinte galletas."
        }

    return {"respuesta": "Di el número y el producto."}